homeassistant/components/teltonika/sensor.py

```diff
@@ -0,0 +1,209 @@
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        suggested_display_precision=0,
+    ),
+    SensorEntityDescription(
+        key="operator",
+        translation_key="operator",
+    ),
+    SensorEntityDescription(
+        key="connection_type",
+        translation_key="connection_type",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: TeltonikaConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Teltonika sensor platform."""
+    data: TeltonikaData = entry.runtime_data
+    coordinator = data.coordinator
+
+    # Track known modems to detect new ones
+    known_modems: set[str] = set()
+
+    @callback
+    def _async_add_new_modems() -> None:
+        """Add sensors for newly discovered modems."""
+        current_modems = set(coordinator.data.keys())
+        new_modems = current_modems - known_modems
+
+        if new_modems:
+            entities = [
+                TeltonikaSensorEntity(
+                    coordinator,
+                    data.device_info,
+                    description,
+                    modem_id,
+                    coordinator.data[modem_id],
+                )
+                for modem_id in new_modems
+                for description in SENSOR_DESCRIPTIONS
+            ]
+            async_add_entities(entities)
+            known_modems.update(new_modems)
+
+    # Add sensors for initial modems
+    _async_add_new_modems()
+
+    # Listen for new modems
+    entry.async_on_unload(coordinator.async_add_listener(_async_add_new_modems))
+
+
+class TeltonikaSensorEntity(CoordinatorEntity, SensorEntity):
+    """Teltonika sensor entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator,
+        device_info,
+        description: SensorEntityDescription,
+        modem_id: str,
+        modem: Any,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._modem_id = modem_id
+        self._attr_device_info = device_info
+
+        # Create unique ID using entry unique identifier, modem ID, and sensor type
+        entry_unique_id = (
+            coordinator.config_entry.unique_id or coordinator.config_entry.entry_id
+        )
+        self._attr_unique_id = f"{entry_unique_id}_{modem_id}_{description.key}"
+
+        # Use translation key for proper naming
+        modem_name = getattr(modem, "name", f"Modem {modem_id}")
+        self._modem_name = modem_name
+        self._attr_translation_key = description.translation_key
+        self._attr_translation_placeholders = {"modem_name": modem_name}
+        self._attr_suggested_object_id = slugify(
+            f"{modem_name} {description.translation_key or description.key}"
+        )
```

Comment:
why do we use translation placeholders here? We also shouldnt have to use suggested object id and let HA generate it